activesupport/lib/active_support/deprecation/behaviors.rb

```diff
@@ -32,12 +32,16 @@ class Deprecation
         logger.debug callstack.join("\n  ") if debug
       },
 
-      notify: ->(message, callstack) {
-        ActiveSupport::Notifications.instrument("deprecation.rails",
-                                                message: message, callstack: callstack)
+      notify: ->(message, callstack, deprecation_horizon, gem_name) {
+        notification_name = "deprecation.#{gem_name.underscore.tr('/', '.')}"
```

Comment:
Maybe `_` instead of `.`? We don't want to mix the gem namespace with the notification namespace. `deprecation.action_controller.test_helper` seems more like a `test_helper` notification namespace instead of the gem namespace.